packages/customize-widgets/src/utils.js

```diff
@@ -98,16 +98,12 @@ export function widgetToBlock( { id, idBase, number, instance } ) {
 	const {
 		encoded_serialized_instance: encoded,
 		instance_hash_key: hash,
-		raw_instance: rawInstance,
+		raw_instance: raw,
 		...rest
 	} = instance;
 
-	// It's unclear why `content` is sometimes `undefined`, but it shouldn't be.
-	const rawContent = rawInstance.content || '';
-	const raw = { ...rawInstance, content: rawContent };
-
 	if ( idBase === 'block' ) {
-		const parsedBlocks = parse( raw.content, {
+		const parsedBlocks = parse( raw.content ?? '', {
```

Comment:
if we're going to revert this change because the block shouldn't have `content`, maybe we add the check in the `if` and avoid calling `parse()` when we know it won't contain anything?

I still feel like this is also an okay workaround, but the implications aren't clear, nor is the reason why some have content and other don't.

the `idBase === 'block'` _must_ be the culprit? by assuming the presence of this `content` property when it shouldn't make that assumption.

Reply:
> if we're going to revert this change because the block shouldn't have `content`, maybe we add the check in the `if` and avoid calling `parse()` when we know it won't contain anything?

That would result in this function creating a `core/legacy-widget` block for a block widget which will display an empty `textarea`, probably alongside some PHP notices because `content` is undefined.

https://github.com/WordPress/wordpress-develop/blob/9ea4e8df9ffaa041fa70fb53b0b8c2ba2a2d0aad/src/wp-includes/widgets/class-wp-widget-block.php#L211

At least with this approach we end up with an empty paragraph block that is useable.

> I still feel like this is also an okay workaround, but the implications aren't clear, nor is the reason why some have content and other don't.

Indeed. Where did this come up? Was it a user bug report? @fullofcaffeine shared the API response but it doesn't explain how the data ended up that way. Seems to me there's invalid data in the database. The proper fix is to prevent that from happening.

Reply:
A nuance of widgets in WordPress is that you can add a widget to a sidebar without ever submitting the widget's form so perhaps something to check is that maybe this bug came from:

1. Activate Classic Widgets plugin and a non-block theme.
2. Go to Appearance → Widgets.
3. Drag Block widget to a sidebar. Don't press save or do anything else.
4. De-activate Classic Widgets plugin and go to Appearance → Customize.

Reply:
> Was it a user bug report? @fullofcaffeine shared the API response but it doesn't explain how the data ended up that way. Seems to me there's invalid data in the database. The proper fix is to prevent that from happening.

Yes, it was originally reported here: https://wordpress.com/forums/topic/sidebar-widget-error/. Unfortunately, we don't know how the content data got into this state :(